pkg/virt-controller/watch/snapshot/snapshot.go

```diff
@@ -656,51 +656,53 @@ func (ctrl *VMSnapshotController) getVolumeSnapshotStatus(vm *kubevirtv1.Virtual
 			Reason:  fmt.Sprintf("Volume is nil [%s]", volume.Name),
 		}
 	}
-	sc, err := ctrl.getVolumeStorageClassForVolume(vm.Namespace, volume)
+	sc, snap, err := ctrl.getVolumeStorageClassForVolume(vm.Namespace, volume)
 	if err != nil {
 		return kubevirtv1.VolumeSnapshotStatus{Name: volume.Name, Enabled: false, Reason: err.Error()}
 	}
-	if sc == "" {
+	if snap == "" {
 		return kubevirtv1.VolumeSnapshotStatus{
 			Name:    volume.Name,
 			Enabled: false,
-			Reason:  fmt.Sprintf("No Volume Snapshot Storage Class found for volume [%s]", volume.Name),
+			Reason:  fmt.Sprintf("No VolumeSnapshotClass: Volume snapshots are not configured for this Storage Class [%s] [%s]", sc, volume.Name),
 		}
 	}
 
 	return kubevirtv1.VolumeSnapshotStatus{Name: volume.Name, Enabled: true}
 }
 
-func (ctrl *VMSnapshotController) getVolumeStorageClassForVolume(namespace string, volume *kubevirtv1.Volume) (string, error) {
+func (ctrl *VMSnapshotController) getVolumeStorageClassForVolume(namespace string, volume *kubevirtv1.Volume) (string, string, error) {
```

Comment:
For me returning the snapClassName from this function is unexpected. Should this be done separately in a different function? It makes the return values a bit confusing as well as this overall method

Reply:
Note it was already returning the snapClassName... just added returning the storageClassName.
Would you accept if we rename it to getVolumeClasses? getVolumeStorageAndSnapshotClasses?

Reply:
@ashleyschuett ^^^ ?